### ISSUE TITLE
Add OOMPolicy=kill to critical-path services

### ISSUE THREAD

**Issue by vonrobak** [vonrobak/fedora-homelab-containers#143]
## Summary
No quadlet sets `OOMPolicy=` or `OOMScoreAdjust=`. Systemd's default (`OOMPolicy=stop`) silently kills the main PID without escalating to unit-level restart logic. For critical services, explicit `OOMPolicy=kill` forces a full-unit kill and clean restart via `Restart=on-failure`.

## Evidence
```
$ grep -l 'OOMPolicy\|OOMScoreAdjust' quadlets/*.container
(no matches)
```

All 31 quadlets rely on systemd's default OOM handling. Under memory pressure, a container's main PID gets SIGKILLed but systemd may not treat this as unit failure — the container enters a confused state where the main process is gone but cgroup lingers.

## Impact
- **Silent partial failures**: OOM kills a child/worker inside the container; main PID survives; systemd sees unit as running; service is actually broken
- **No automatic recovery**: Without `OOMPolicy=kill`, the unit may not transition to `failed` state, so `Restart=on-failure` doesn't trigger
- **Diagnostic noise**: `journalctl` shows the OOM kill but unit status reads "active (running)" — misleading during incidents

## Remediation
Add to the `[Service]` (or `[Container]` — whichever section applies under quadlet semantics) block of critical services:

```
OOMPolicy=kill
```

Target services (in priority order):
- `traefik.container` — edge, no traffic = full outage
- `authelia.container` — SSO failure cascades to all protected services
- `prometheus.container` — monitoring blindness = blind system
- `nextcloud-db.container` — partial-write DB corruption risk
- `postgresql-immich.container` — same
- `redis-authelia.container` — session loss = logout storm

Apply:
```bash
systemctl --user daemon-reload
systemctl --user restart traefik authelia prometheus nextcloud-db postgresql-immich redis-authelia
```

## Verification
```bash
# Confirm the directive is active
for svc in traefik authelia prometheus; do
  systemctl --user show -p OOMPolicy $svc.service
done
# Expect: OOMPolicy=kill

# Optional: simulate OOM on a test container (NOT on one of the above)
# Create a test unit with MemoryMax=64M, run stress-ng --vm 1 --vm-bytes 100M
# Confirm unit transitions to "failed" and auto-restarts
```

## Effort
~20 min total (edit 6 files, daemon-reload, restart).

## Context
2026-04-17 service configuration review §4 MEDIUM-10. Relates to MEDIUM-11 (CPUQuota matrix) — consider landing them together as one "resource governance" PR.
